### PR TITLE
Add named pipes and shared memory connections

### LINKED ISSUE #187: Update go-mssqldb and support named pipes and shared memory
go-mssqldb now supports named pipes, so go-sqlcmd can too.
Will try to get it in to v1

### PR DESCRIPTION
**Pull request by shueybubbles**
Fixes #187 
Now that `go-mssqldb` supports `np` and `lpc` protocols, so can go-sqlcmd, at least on Windows.

### CODE REVIEW

File: .vscode/launch.json [hunk 1]
```diff
@@ -24,16 +25,16 @@
             "type" : "go",
             "request": "launch",
             "mode" : "auto",
-            "program": "${workspaceFolder}/cmd/sqlcmd",
-            "args" : ["-Q", "EXIT(select 100 as Count)"],
+            "program": "${workspaceFolder}/cmd/modern",
+            "args" : ["-Q", "EXIT(select network_protocol from sys.dm_exec_sessions)", "-S", "lpc:."],
```
Comment (apoorvdeshmukh) [inline .vscode/launch.json:29]: Do we already have a unit test to test this scenario?

File: .vscode/launch.json [hunk 2]
```diff
@@ -24,16 +25,16 @@
...
         },
         {
             "name" : "Run file query",
             "type" : "go",
             "request": "launch",
             "mode" : "auto",
-            "program": "${workspaceFolder}/cmd/sqlcmd",
-            "args" : ["-i", "testdata\\select100.sql"],
+            "program": "${workspaceFolder}/cmd/modern",
+            "args" : ["-S", "np:.", "-i", "${workspaceFolder}/cmd/sqlcmd/testdata/select100.sql"],
```
Comment (apoorvdeshmukh) [inline .vscode/launch.json:37]: Do we need to turn this into unit test?

**Review by apoorvdeshmukh — APPROVED** [review]
Looks good!

**Comment by stuartpa** [comment]
I'm 8 getting unit test failures on my local machine with this branch

=== RUN   TestExitOnError
unable to get instances from Sql Server Browser on host localhost: read udp [::1]:63086->[::1]:1434: i/o timeout
    sqlcmd_test.go:286: 
        	Error Trace:	C:\src\go-sqlcmd\cmd\sqlcmd\sqlcmd_test.go:286
        	
I run with Browser disabled.  If I enable browser, then the tests don't fail, but this PR should not require browser service be running moving forward.

**Review by stuartpa — CHANGES_REQUESTED** [review]
I'm 8 getting unit test failures on my local machine with this branch

=== RUN TestExitOnError
unable to get instances from Sql Server Browser on host localhost: read udp [::1]:63086->[::1]:1434: i/o timeout
sqlcmd_test.go:286:
Error Trace: C:\src\go-sqlcmd\cmd\sqlcmd\sqlcmd_test.go:286

I run with Browser disabled. If I enable browser, then the tests don't fail, but this PR should not require browser service be running moving forward.

We need a unit test to catch this in the future (make sure we don't approve a change that causes browser service running to be a dependency)

**Comment by stuartpa** [comment]
net stop "SQL Server Browser" 

to stop it

**Comment by shueybubbles** [comment]
the browser dependency will be fixed by tagging go-mssqldb to v0.20.0 and using that build, which has the fix.

**Comment by shueybubbles** [comment]
I'm going to sit on this PR for a while and work on some higher priority items (like NOTICE generation) and come back to it. 
I'm encountering some problems with connections going idle over named pipes and not being handled correctly, and I need to figure out if it's something go-mssqldb can handle or if sqlcmd isn't handling the error correctly.

File: pkg/sqlcmd/sqlcmd.go [hunk 3]
```diff
@@ -534,3 +536,12 @@ func (s Sqlcmd) Log(_ context.Context, _ msdsn.Log, msg string) {
 	_, _ = s.GetOutput().Write([]byte("DRIVER:" + msg))
 	_, _ = s.GetOutput().Write([]byte(SqlcmdEol))
 }
+
+func init() {
+	if len(msdsn.ProtocolParsers) == 3 {
+		// reorder the protocol parsers to lpc->np->tcp
```
Comment (stuartpa) [inline pkg/sqlcmd/sqlcmd.go:542]: Can you include the why in this comment.  I don't know why this is being done.

File: .vscode/launch.json (hunk 2 above)
Comment (shueybubbles) [reply .vscode/launch.json:37]: the tests exist. The launch.json is needed to debug the full app starting in main.go

**Pull request merged**